ui/src/app/views/account-creation-modals/component.ts

```diff
@@ -0,0 +1,64 @@
+import {Component, EventEmitter, Input, OnInit, Output} from '@angular/core';
+
+import {ProfileService} from 'generated';
+import {ResendWelcomeEmailRequest, UpdateContactEmailRequest} from 'generated';
+import {ServerConfigService} from '../../services/server-config.service';
+
+@Component({
+  selector: 'app-account-creation-modals',
+  templateUrl: './component.html',
+  styleUrls: ['./component.css',
+    '../../styles/buttons.css'],
+})
+export class AccountCreationModalsComponent implements OnInit {
+  changingEmail = false;
+  resendingEmail = false;
+  contactEmail: string;
+  @Input('username') username: string;
+  @Input('gsuiteDomain') gsuiteDomain: string;
+
+  @Output() updateEmail = new EventEmitter<string>();
+
+  constructor(
+    private profileService: ProfileService,
+    serverConfigService: ServerConfigService
+  ) {
+    serverConfigService.getConfig().subscribe((config) => {
+      this.gsuiteDomain = config.gsuiteDomain;
+    });
+    this.contactEmail = '';
+  }
+
+  ngOnInit() {
+  }
+
+  updateAndSendEmail() {
+    this.changingEmail = true;
+    this.contactEmail = '';
+  }
+
+  resendInstructions() {
+    this.resendingEmail = true;
+  }
+
+  sendAndUpdate() {
+    const request: UpdateContactEmailRequest = {
+      username: this.username + '@' + this.gsuiteDomain,
+      contactEmail: this.contactEmail
+    };
+    this.updateEmail.emit(this.contactEmail);
+    this.profileService.updateContactEmail(request).subscribe(() => {
+      this.changingEmail = false;
```

Comment:
Do we need a loading indicator while this XHR is pending? e.g. clrLoading on the send button